shared-data/python/opentrons_shared_data/types.py

```diff
@@ -0,0 +1,24 @@
+"""Shared-data types."""
+
+from typing import Generic, TypeVar
+from pydantic import (
+    BaseModel,
+    StrictInt,
+    StrictFloat,
+    NonNegativeInt,
+    NonNegativeFloat,
+)
+
+
+Number = StrictInt | StrictFloat
+NonNegativeNumber = NonNegativeInt | NonNegativeFloat
+
+NumberType = TypeVar("NumberType")
+
+
+class Vec3f(BaseModel, Generic[NumberType]):
+    """A 3D Vector."""
+
+    x: NumberType
+    y: NumberType
+    z: NumberType
```

Comment:
1. The name `Vec3f` comes from "vector of 3 `float`s." If this is generic across other kinds of `NumberType`, let's rename it to just `Vec3`.
2. Since we're really pushing this deep down into common shared code, I'm worried that someone will come along later and add an `= 0` default for something without realizing that it will affect a million things. Let's spell out that this is specifically for vectors where all elements are required, to hopefully prompt people to use a different type if that's not the behavior that they want.

```suggestion
class Vec3f(BaseModel, Generic[NumberType]):
    """A 3D vector where all elements are required."""

    x: NumberType
    y: NumberType
    z: NumberType
```

Reply:
Reverted these vector refactors in fe5005e. We can un-revert them in another PR.